red-hat-data-services/odh-dashboard
# RHODS-2856: Opt in user data collection.

- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message - N/A
- [x] JIRA link(s):   https://issues.redhat.com/browse/RHODS-2856

This PR adds the ability for a RHODS admin to enable and disable user interaction data collection. 

Testing instructions:

RHODS admin testing:
1. Login to the RHODS dashboard as a RHODS admin.
2. Verify that Settings in the left hand navigation is present.
3. Verify that Cluster Settings is a menu item under Settings.
4. Navigate to Cluster Settings.
5. Verify that the "Allow collection of usage data" is checked or unchecked, based on redhat-ods-applications --> rhods-segment-key-config --> segmentKeyEnabled Config map in OpenShift.
6. Check or uncheck "Allow collection of usage data"
7. Verify that the config map in step 5 is updated.

<img width="1797" alt="Screen Shot 2022-02-21 at 9 12 43 AM" src="https://user-images.githubusercontent.com/6126356/154971869-657e42fa-dd0f-4bcd-813c-a72cb5551144.png">

RHODS non admin testing:
1. Login to the RHODS dashboard as a non RHODS admin (RHODS developer).
2. Verify that Settings in the left hand navigation is NOT present.
3. Verify that Cluster Settings is NOT present.
